### PR TITLE
InterOP: enable test report generation for tests

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

**What type of PR is this:**

/kind tests

**What does this PR do / why we need it:**
As we are want to start InterOP testing for v3, we need to start test reports generation.

**Which issue(s) this PR fixes:**

Fixes #

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Review by rm3l — CHANGES_REQUESTED** [review]
LGTM, but can you add all those XML report files to `.gitignore` as well? This will prevent committing them accidentally to the repo after running those `make` targets.

/approve

File: .gitignore [hunk 1]
```diff
@@ -14,7 +14,7 @@ bin
 coverage.txt
 
 # Ignore test reports
-reports/
+test-*.xml
```
Comment (rm3l) [inline .gitignore:17]: I guess nothing creates this `reports` directory? I saw an occurrence of this in `scripts/run_script_e2e.sh`, but I guess that script is no longer used, right?

https://github.com/redhat-developer/odo/blob/5a39eda50e0e335cb063b7748dae8302014ca2bc/scripts/run_script_e2e.sh#L88

Comment (anandrkskd) [reply .gitignore:17]: No, it was previously used for test on PSI

**Review by rm3l — APPROVED** [review]

**Comment by rm3l** [comment]
```
    Deleting dir: C:\Users\Administrator.ANSIBLE-TEST-VS\AppData\Local\Temp\2777555636
    Unable to delete C:\Users\Administrator.ANSIBLE-TEST-VS\AppData\Local\Temp\2777555636 on attempt #4, trying again...
    [odo] I1116 04:58:59.493023    2412 delete.go:80] failed to delete resource "binding-zdpp" (binding.operators.coreos.com.v1alpha1.servicebindings): timeout while waiting for "binding-zdpp" resource to be deleted
    [odo] Failed to delete the "Deployment" resource: mynode-app
    [odo] Failed to delete the "ServiceBinding" resource: binding-zdpp
    Deleting dir: C:\Users\Administrator.ANSIBLE-TEST-VS\AppData\Local\Temp\2777555636
    Successfully deleted C:\Users\Administrator.ANSIBLE-TEST-VS\AppData\Local\Temp\2777555636 after #5 attempts
    Deleting dir: C:\Users\Administrator.ANSIBLE-TEST-VS\AppData\Local\Temp\1672953641
  << End Captured GinkgoWriter Output

  Timed out after 180.003s.
  Expected process to exit.  It did not.
  In [BeforeEach] at: C:/Users/Administrator.ANSIBLE-TEST-VS/2486/tests/helper/helper_dev.go:187
------------------------------
+++++++
------------------------------
+ [SLOW TEST] [121.536 seconds]
odo dev command tests when a hotReload capable project is used with odo dev should execute the run command
C:/Users/Administrator.ANSIBLE-TEST-VS/2486/tests/integration/cmd_dev_test.go:2464
------------------------------


Summarizing 1 Failure:
  [FAIL] odo add binding command tests when the component is bootstrapped when adding a binding (current namespace) when odo dev is run when odo dev command is stopped [BeforeEach] should have successfully delete the binding
  C:/Users/Administrator.ANSIBLE-TEST-VS/2486/tests/helper/helper_dev.go:187

Ran 325 of 399 Specs in 902.445 seconds
FAIL! -- 324 Passed | 1 Failed | 0 Pending | 74 Skipped
```

Retriggering the tests...

**Comment by rm3l** [comment]
/override ci/prow/v4.11-integration-e2e

Tests passing on IBM Cloud.

**Pull request merged**